addons/mail/controllers/discuss.py

```diff
@@ -432,6 +432,8 @@ def channel_ping(self, channel_id, rtc_session_id=None, check_rtc_session_ids=No
     @http.route('/mail/thread/data', methods=['POST'], type='json', auth='user')
     def mail_thread_data(self, thread_model, thread_id, request_list, **kwargs):
         thread = request.env[thread_model].with_context(active_test=False).search([('id', '=', thread_id)])
+        if not thread:
+            return {'hasReadAccess': False}
```

Comment:
This should be moved into `_get_mail_thread_data` with all the other values. move `res` and quick return above the ensure one